ksql-functional-tests/src/test/resources/query-validation-tests/ensure-formatter.json

```diff
@@ -0,0 +1,29 @@
+{
+  "comments": [
+    "Query Translation tests should make sure that the command that is being executed can be ",
+    "persisted properly onto the command topic (i.e. serialized using SqlFormatter). Otherwise, ",
+    "it is possible that we may not execute the same statement after enqueuing as we did before. ",
+    "This test will fail if we refactor QueryTranslationTest in a way that makes it no longer go ",
+    "through the SqlFormatInjector, which serializes and then deserializes the statements to make ",
+    "sure that these inconsistencies are caught.",
+    "",
+    "NOTE: This test is a little hacky at the moment because it relies on us not formatting SET ",
+    "expressions. There is nothing preventing us in the future from doing this, but I don't have ",
+    "a better way as of now to ensure failure."
+  ],
+  "tests": [
+    {
+      "name": "ensure QTTs go through SqlFormatter",
+      "statements": [
+        "CREATE STREAM IGNORED (ID bigint) WITH (kafka_topic='topic', value_format='DELIMITED');",
+        "SET 'auto.offset.reset'='earliest';"
+      ],
+      "inputs": [ ],
+      "outputs": [ ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Unable to format statement!"
```

Comment:
I could not understand why this could fail. Could you explain in the comment what can go wrong that would make this to fail? I see a CREATE command which will be persisted in the command topic. How can this fail if the SqlFormatter is not called?

Reply:
it's a little hacky, but `SET` calls fail to format (they're not supported in `SqlFormatter`). This test makes sure that a failure _does_ happen. Without running through the `SqlFormatter`, a different failure happens - not ("Unable to format statement!")